File: Programming_Languages/C/Node.c
#include <stdio.h>
#include <stdlib.h>
struct node {
	int data;
	struct node* left;
	struct node* right;
};

struct node* newNode(int data)
{
	// Allocate memory for new node
	struct node* node
		= (struct node*)malloc(sizeof(struct node));

	// Assign data to this node
	node->data = data;

	// Initialize left and
	// right children as NULL
	node->left = NULL;
	node->right = NULL;
	return (node);
}
//Traversal [Preorder]
void preOrder(Node*root){
  if(root==NULL){
    return;
  }
  printf(root->data);
  preOrder(root->left);
  preOrder(root->right);
}
int main()
{
	/*create root*/
	struct node* root = newNode(1);
	root->left = newNode(2);
	root->right = newNode(3);
	root->left->left = newNode(4);
	getchar();

    //Print
    preOrder(root)
	return 0;
}


//Contributed by KmAlan

File: Programming_Languages/Java/Node.java
class Node{
    int data;
    Node left,right;
    Node(data){
        this.data=data;
        this.left =this.right = null;
    }

}
class BinaryTree
{
	// Root of Binary Tree
	Node root;

	// Constructors
	BinaryTree(int key)
	{
		root = new Node(key);
	}

	BinaryTree()
	{
		root = null;
	}

    //Tree Traversal[Preorder]
    public void preOrder(Node root,List<Integer> result){
        if (root == null){
            return;
        }
        res.add(root.data);
        //Recursive call
        preOrder(root.left);
        preOrder(root.right);
    }
	public static void main(String[] args)
	{
		BinaryTree tree = new BinaryTree();

		/*create root*/
		tree.root = new Node(1);
		tree.root.left = new Node(2);
		tree.root.right = new Node(3);
		tree.root.left.left = new Node(4);

        //List to store results
        List<Integer> result = new ArrayList<>();
        tree.preOrder(root,result);


        //Print results
        System.out.print(result)
}


//Contibuted by KmAlan